subworkflows/local/bam_downsample/main.nf

```diff
@@ -61,25 +57,30 @@ workflow BAM_DOWNSAMPLE {
     ch_bam_emul = SAMTOOLS_VIEW.out.bam
         .combine(SAMTOOLS_INDEX_1.out.bai, by:0)
 
-    SAMTOOLS_MERGE(
-        ch_bam_emul
-            .map{
-                metaICRD, bam, index -> [metaICRD.subMap("id", "depth"), bam, index]
-            }
-            .groupTuple()
-            .map{ metaID, bam, index ->
-                [ metaID + ["chr": "all"], bam, index ]
-            },
-        ch_fasta
-    )
-    SAMTOOLS_INDEX_2(SAMTOOLS_MERGE.out.bam)
-    ch_versions = ch_versions.mix(SAMTOOLS_INDEX_2.out.versions.first())
+    if (params.sim_by_reg == true) {
```

Comment:
Similar suggestion:

Instead of adding a new params, is it possible to evaluate if the user thas set `--input_region != null`? I think our current behavior is that if the user does not add anything, then we process by full bam. I think in this case this is similar, and we can simplify by having less params.